prospero-common/src/main/java/org/wildfly/prospero/api/SavedState.java

```diff
@@ -79,20 +93,96 @@ public String getMsg() {
         return msg;
     }
 
+    public List<Version> getManifestVersions() {
+        return manifestVersions;
+    }
+
     public String shortDescription() {
-        return String.format("[%s] %s - %s %s", hash, timestamp.toString(), type.toString().toLowerCase(Locale.ROOT), this.msg);
+        final String msg;
+        if (manifestVersions.isEmpty()) {
+            msg = this.msg;
+        } else {
+            final String versions = manifestVersions.stream().map(Version::getVersion).collect(Collectors.joining("+"));
+            msg = "[" + versions + "]";
+        }
+        return String.format("[%s] %s - %s %s", hash, timestamp.toString(), type.toString().toLowerCase(Locale.ROOT), msg);
     }
 
     @Override
     public boolean equals(Object o) {
         if (this == o) return true;
         if (o == null || getClass() != o.getClass()) return false;
         SavedState that = (SavedState) o;
-        return Objects.equals(hash, that.hash) && Objects.equals(timestamp, that.timestamp) && type == that.type;
+        return Objects.equals(manifestVersions, that.manifestVersions) && Objects.equals(hash, that.hash) && Objects.equals(timestamp, that.timestamp) && type == that.type && Objects.equals(msg, that.msg);
```

Comment:
It's probably not very important, but since the manifest versions are involved in the equals method, should the collection type for versions be Set, rather than List?

(Thinking purely about ordering, which I assume is generally undefined - version collections should be equal if they contain the same elements, irrespective of order. I think Set would facilitate that.)